documentation/specs/dotnet-run-for-maui.md

```diff
@@ -0,0 +1,221 @@
+    (workload), it can print a friendly message and exit.
+
+* `dotnet run --list-targets` will then basically exit early, never
+  running any build, deploy, `ComputeRunArguments`, or run steps.
+
+A new `--target|-t` switch will:
+
+* bypass the target-selection portion of the `run` workflow described above
+
+* Pass in the `-p:RunTarget` global MSBuild property to all build,
+  deploy, `ComputeRunArguments`, or run steps.
+
+* The iOS and Android workloads will know how to interpret `$(RunTarget)`
+  to select an appropriate device, emulator, or simulator.
+
+## What about Launch Profiles?
+
+The iOS and Android workloads ignore all
+`Properties/launchSettings.json` files and do nothing with them.
+
+WindowsAppSDK requires a launch profile to select between "packaged"
+MSIX and an "unpackaged" .exe, and so we currently provide this in the
+`dotnet new maui` project template:
+
+```json
+{
+  "profiles": {
+    "Windows Machine": {
+      "commandName": "Project",
+      "nativeDebugging": false
+    }
+  }
+}
+```
+
+Or if you want to be "packaged":
+
+```json
+{
+  "profiles": {
+    "Windows Machine": {
+      "commandName": "MsixPackage",
+      "nativeDebugging": false
+    }
+  }
+}
+```
+
+Launch profiles don't immediately look useful for iOS or Android, as
+the identifier you'd put in here would be different per developer --
+you wouldn't want the value saved in source control. You could put a
+generic selection like device vs emulator/simulator, but that isn't
+addressing the problem we are interested in. Full launch profile
+support for .NET MAUI projects may be interesting to address in the
+future.
+
+## iOS and Android workload behavior
+
+We will work to align as much behavior as possible between the
+platforms. Most of this work will happen in the dotnet/android and
+dotnet/macios repo, and would be orthogonal to the changes in the .NET
+SDK.
+
+## macOS and MacCatalyst Behavior
+
+`dotnet run` "just works" on macOS Desktop, because `$(RunCommand)`
+can launch the app directly -- similar to console apps.
+
+## WindowsAppSDK Behavior
+
+Running a `dotnet new maui` project works today because of the default
+launch profile in "unpackaged" mode:
+
+```dotnetcli
+> dotnet run -f net10.0-windows10.0.19041.0
+Using launch settings from D:\src\hellomaui\Properties\launchSettings.json...
+```
+
+To improve the behavior for "packaged" / MSIX, we could:
+
+* Implement the `DeployToRunTarget` MSBuild target.
+
+* Implement the `ComputeRunArguments` MSBuild target.
+
+In the future, we can either add this logic into the .NET MAUI
+workload or WindowsAppSDK itself.
+
+## Other frameworks: Avalonia, Uno, MonoGame, etc.
+
+When these frameworks run on iOS or Android, they are basically using
+the `ios` and `android` workloads *without* .NET MAUI. Any iOS or
+Android-specific behavior would apply to these project types in the
+same way a `dotnet new android` or `dotnet new ios` project template
+would.
```

Comment:
@baronfel these frameworks should act like a plain iOS or Android project -- and should "just work".

The only platform we should probably look into is wasm and if there are any issues there.